src/azure-cli/azure/cli/command_modules/keyvault/_params.py

```diff
@@ -164,6 +164,10 @@ def load_arguments(self, _):
 
     with self.argument_context('keyvault key set-attributes') as c:
         c.attributes_argument('key', KeyAttributes)
+
+    for scope in ['list', 'list-deleted', 'list-versions']:
+        with self.argument_context('keyvault key {}'.format(scope)) as c:
+            c.argument('maxresults', options_list=['--maxresults'], type=int)
```

Comment:
is it necessary to set a upper limit on maxresults?

Reply:
@yungezz We have already set the lower bound and upper bound of `maxresults` in swagger and SDK code, it's not necessay to set it again in CLI code.

Swagger code: https://github.com/Azure/azure-rest-api-specs/blob/84ab02cad8cb3d3c97118e989815dbe5c7d5b529/specification/keyvault/data-plane/Microsoft.KeyVault/stable/7.0/keyvault.json#L1179-L1189

SDK code: https://github.com/Azure/azure-sdk-for-python/blob/657d9c8e6cbb8bd543d2996112ee9f00dd45cf87/sdk/keyvault/azure-keyvault-certificates/azure/keyvault/certificates/_shared/_generated/v7_0/operations/_key_vault_client_operations.py#L1743-L1744

Reply:
What would happen if maxresults passed in is greater than 25?

Reply:
@fengzhou-msft CLI will throw an error message:

```
(env3) D:\repos\azure-cli>az keyvault secret list --vault-name bim-foo --maxresults 100
Parameter 'maxresults' must be less than 25.
```

Reply:
Ok. At least users know the limit from the error message. If you could put it in help text, that will be more clear, just a suggestion though.

Reply:
@fengzhou-msft Yes, I agree. I think it's a better idea that the upcoming CLI codegen system auto-generates such type of help messages from swagger file.

Reply:
Good to know that you will collect all these feedbacks and backpropagate to CodeGen team.